test/modules/TestAnalyzer/TestResultReporter.cs

```diff
@@ -101,16 +101,46 @@ static ModuleMessagesReport GetReceivedMessagesReport(string moduleId, double to
                 }
             }
 
-            // check if last message is older
-            if (DateTime.Compare(lastMessageDateTime.AddMilliseconds(toleranceInMilliseconds), endDateTime) < 0)
+            bool areMessagesMissing = missingCounter > 0;
+            bool areLatestMessagesTooOld = DateTime.Compare(lastMessageDateTime.AddMilliseconds(toleranceInMilliseconds), endDateTime) < 0;
+            string missingMessagesStatus = $"Missing messages: {missingCounter}. ";
+            string messagesTooOldStatus = $"No messages received for the past {toleranceInMilliseconds} milliseconds. ";
+            string noMissingMessagesStatus = "All messages received. ";
+
+            if (areMessagesMissing)
+            {
+                Logger.LogInformation($"Module {moduleId}: missing messages");
+            }
+
+            if (areLatestMessagesTooOld)
             {
                 Logger.LogInformation($"Module {moduleId}: last message datetime={lastMessageDateTime} and end datetime={endDateTime}");
-                return new ModuleMessagesReport(moduleId, StatusCode.OldMessages, totalMessagesCounter, $"Missing messages: {missingCounter}. No messages received for the past {toleranceInMilliseconds} milliseconds.", lastMessageDateTime, missedMessages, Settings.Current.TestInfo);
             }
 
-            return missingCounter > 0
-                ? new ModuleMessagesReport(moduleId, StatusCode.SkippedMessages, totalMessagesCounter, $"Missing messages: {missingCounter}.", lastMessageDateTime, missedMessages, Settings.Current.TestInfo)
-                : new ModuleMessagesReport(moduleId, StatusCode.AllMessages, totalMessagesCounter, "All messages received.", lastMessageDateTime, Settings.Current.TestInfo);
+            StatusCode statusCode;
```

Comment:
General comments: We put more than 1 information in status code.  If we needed to consider 1 more information to determine status code, it will make status code assignment logic more complicated.  We go for this quick approach now as more changes are required in iotedge dashboard if we don't provide status code.  Good for now.